lib/fizzy/parser_expr.cpp

```diff
@@ -799,14 +805,27 @@ parser_result<Code> parse_expr(const uint8_t* pos, const uint8_t* end, FuncIdx f
             push(code.immediates, static_cast<uint32_t>(value));
             break;
         }
-
         case Instr::i64_const:
         {
             int64_t value;
             std::tie(value, pos) = leb128s_decode<int64_t>(pos, end);
             push(code.immediates, static_cast<uint64_t>(value));
             break;
         }
+        case Instr::f32_const:
+        {
+            uint32_t value;
+            std::tie(value, pos) = read<decltype(value)>(pos, end);
```

Comment:
what is the point of using `decltype` here if `value` is declared on the previous line?

Reply:
Easier to copy-paste to `Instr::f64_const`.